src/common/layouts/sidebar/variants/Courses.tsx

```diff
@@ -51,11 +49,11 @@ export interface TopBarProps {
 }
 
 export function CoursesTopBar({ onFilter }: TopBarProps): JSX.Element {
-  const { isOpen, onToggle } = useDisclosure();
   const { term } = useParams();
   const location = useLocation();
   const [searchParams] = useSearchParams();
   const mode = useDarkMode();
+  const [filter] = useSessionStorage<boolean>('filter_courses', true);
```

Comment:
I misguided you :bow: 
You'll want to remove this and add a property called `filter` on `TopBarProps` here instead.
Then you'll want to pass the `filter` value into `<CourseTopBar/>` on line 134. 

That way we avoid having multiple uses of hooks and the switch button remains in sync with the session storage value :)

Reply:
While this would have worked the same, this is a bit of a cleaner way.